src/model/calculations.ts

```diff
@@ -100,11 +101,154 @@ export class OptionsRemainingCalculator {
   }
 }
 
+interface StockClass {
+  object_type: string;
+  id: string;
+  name: string;
+  class_type: string;
+  default_id_prefix?: string;
+  current_shares_authorized?: string;
+  board_approval_date?: string;
+  votes_per_share?: number;
+  conversion_rights?: StockClassConversionRight[];
+}
+
+interface StockClassConversionRight {
+  conversion_mechanism: {
+    type: string;
+    ratio: {
+      numerator: string;
+      denominator: string;
+    };
+  };
+  rounding_type?: string;
+  converts_to_stock_class_id?: string;
+}
+
+export class ConversionRatioCalculator {
+  private stockClasses: StockClass[] = [];
+
+  public findRatio(preferredStockClassId: string): {
```

Comment:
**Question:** Why not pass in the StockClass instead of just the id?

Reply:
I don't think that simplifies the interface; I want to deal in objects. If I have a StockClass then I want to either ask the StockClass what its ratio is, or ask something else (this calculator) what the conversion ratio on my StockClass is.

Reply:
I'll do that in the next PRs